api/src/main/java/org/eclipse/microprofile/config/spi/ConfigValueSource.java

```diff
@@ -0,0 +1,185 @@
+     * <p>
+     *
+     * The returned property names may be a subset of the names of the total set of retrievable properties in this
+     * config source.
+     * <p>
+     *
+     * The returned set is not required to allow concurrent or multi-threaded iteration; however, if the same set is
+     * returned by multiple calls to this method, then the implementation must support concurrent and multi-threaded
+     * iteration of that set.
+     * <p>
+     *
+     * The set of keys returned <em>may</em> be a point-in-time snapshot, or <em>may</em> change over time (even
+     * during active iteration) to reflect dynamic changes to the available set of keys.
+     *
+     * @return a set of property names that are known to this configuration source
+     */
+    Set<String> getPropertyNames();
+
+    /**
+     * The name of the configuration source. The name might be used for logging or for analysis of configured values,
+     * and also may be used in {@linkplain #getOrdinal() ordering decisions}.
+     * <p>
+     *
+     * An example of a configuration source name is "{@code property-file mylocation/myprops.properties}".
+     *
+     * @return the name of the configuration source
+     */
+    String getName();
+
+    /**
+     * Return the ordinal priority value of this configuration source.
+     * <p>
+     *
+     * If a property is specified in multiple config sources, the value in the config source with the highest ordinal
+     * takes precedence. For configuration sources with the same ordinal value, the configuration source name will
+     * be used for sorting according to string sorting criteria.
+     * <p>
+     *
+     * Note that this method is only evaluated during the construction of the configuration, and does not affect
+     * the ordering of configuration sources within a configuration after that time.
+     * <p>
+     *
+     * The ordinal values for the default configuration sources can be found
+     * <a href="#default_config_sources">above</a>.
+     *
+     * <p>
+     * Any configuration source which is a part of an application will typically use an ordinal between 0 and 200.
+     * Configuration sources provided by the container or 'environment' typically use an ordinal higher than 200.
+     * A framework which intends have values overridden by the application will use ordinals between 0 and 100.
+     *
+     * <p>
+     * The default implementation of this method looks for a configuration property named
+     * "{@link #CONFIG_ORDINAL config_ordinal}" to determine the ordinal value for this configuration source. If the
+     * property is not found, then the {@linkplain #DEFAULT_ORDINAL default ordinal value} is used.
+     * <p>
+     *
+     * This method may be overridden by configuration source implementations to provide a different behavior.
+     *
+     * @return the ordinal value
+     */
+    default int getOrdinal() {
```

Comment:
Referencing #553 here.

Reply:
Yes, this is just as is copy from the original `ConfigSource`. I'm in favour of #553 for both.